lnrpc/walletrpc/driver.go

```diff
@@ -0,0 +1,77 @@
+// +build walletrpc
+
+package walletrpc
+
+import (
+	"fmt"
+
+	"github.com/lightningnetwork/lnd/lnrpc"
+)
+
+// createNewSubServer is a helper method that will create the new WalletKit RPC
+// sub server given the main config dispatcher method. If we're unable to find
+// the config that is meant for us in the config dispatcher, then we'll exit
+// with an error.
+func createNewSubServer(configRegistry lnrpc.SubServerConfigDispatcher) (lnrpc.SubServer, lnrpc.MacaroonPerms, error) {
+	// We'll attempt to look up the config that we expect, according to our
+	// subServerName name. If we can't find this, then we'll exit with an
+	// error, as we're unable to properly initialize ourselves without this
+	// config.
+	walletKitServerConf, ok := configRegistry.FetchConfig(subServerName)
+	if !ok {
+		return nil, nil, fmt.Errorf("unable to find config for "+
+			"subserver type %s", subServerName)
+	}
+
+	// Now that we've found an object mapping to our service name, we'll
+	// ensure that it's the type we need.
+	config, ok := walletKitServerConf.(*Config)
+	if !ok {
+		return nil, nil, fmt.Errorf("wrong type of config for "+
+			"subserver %s, expected %T got %T", subServerName,
+			&Config{}, walletKitServerConf)
+	}
+
+	// Before we try to make the new WalletKit service instance, we'll
+	// perform some sanity checks on the arguments to ensure that they're
+	// useable.
+	switch {
+	case config.MacService == nil:
+		return nil, nil, fmt.Errorf("MacService must be set to " +
+			"create WalletKit RPC server")
```

Comment:
It seems walletkit cannot be used with nomacaroons

Reply:
Fixed!